keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -71,6 +73,36 @@ class ArtifactListener(
     }
   }
 
+  /**
+   * For each registered debian artifact, get the last version, and persist if it's newer than what we have.
+   */
+  // todo eb: this doesn't need to run every couple of minutes on each instance...
+  // todo eb: should we fetch more than one version?
+  @Scheduled(fixedDelayString = "\${keel.artifact-refresh.frequency:PT3M}")
+  fun syncDebArtifactVersions() =
+    runBlocking {
+      artifactRepository.getAll(DEB).forEach { artifact ->
+        val lastRecordedVersion: String? = artifactRepository.versions(artifact).firstOrNull()
```

Comment:
Instead of doing the whole thing inside of a `runBlocking` scope you could put `launch` inside the loop which will create a new coroutine for each artifact. We might want to consider backing off or scheduling checks for each artifact individually once we start handling a lot, though.

Reply:
Ok. I'll do that, bump the scheduled method way down, and add an admin endpoint to trigger it. That seems reasonable.